packages/suite/src/hooks/wallet/useStakeEthForm.ts

```diff
@@ -0,0 +1,332 @@
+    CRYPTO_INPUT,
+    OUTPUT_AMOUNT,
+} from 'src/types/wallet/stakeEthForm';
+import {
+    getComposeAddressPlaceholder,
+    mapTestnetSymbol,
+} from 'src/utils/wallet/coinmarket/coinmarketUtils';
+import { useFormDraft } from './useFormDraft';
+import type { AppState } from 'src/types/suite';
+import { AmountLimits } from 'src/types/wallet/coinmarketCommonTypes';
+
+import { fromWei } from 'web3-utils';
+import { useStakeEthFormDefaultValues } from './useStakeEthFormDefaultValues';
+import { useCompose } from './form/useCompose';
+import { MIN_ETH_AMOUNT_FOR_STAKING, MIN_ETH_FOR_FEES } from 'src/constants/suite/ethStaking';
+import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
+
+export const StakeEthFormContext = createContext<StakeEthContextValues | null>(null);
+StakeEthFormContext.displayName = 'StakeEthFormContext';
+
+const useStakeEthState = (
+    selectedAccount: UseStakeEthFormProps['selectedAccount'],
+    fees: AppState['wallet']['fees'],
+    currentState: boolean,
+    defaultFormValues?: StakeEthFormState,
+) => {
+    // do not calculate if currentState is already set (prevent re-renders)
+    if (selectedAccount.status !== 'loaded' || currentState) return;
+
+    const { account, network } = selectedAccount;
+    const coinFees = fees[account.symbol];
+    const levels = getFeeLevels(account.networkType, coinFees);
+    const feeInfo = { ...coinFees, levels };
+
+    return {
+        account,
+        network,
+        feeInfo,
+        formValues: defaultFormValues,
+    };
+};
+
+export const useStakeEthForm = ({
+    selectedAccount,
+}: UseStakeEthFormProps): StakeEthContextValues => {
+    const dispatch = useDispatch();
+
+    const accounts = useSelector(state => state.wallet.accounts);
+    const device = useSelector(selectDevice);
+    const fiat = useSelector(state => state.wallet.fiat);
+    const localCurrency = useSelector(selectLocalCurrency);
+    const fees = useSelector(state => state.wallet.fees);
+
+    const { account, network } = selectedAccount;
+    const { symbol } = account;
+
+    const symbolForFiat = mapTestnetSymbol(symbol);
+    const fiatRates = fiat.coins.find(item => item.symbol === symbolForFiat);
+    // TODO: Implement fee switcher
+    const selectedFee = 'normal';
+
+    const amountLimits: AmountLimits = {
+        currency: symbol,
+        minCrypto: MIN_ETH_AMOUNT_FOR_STAKING.toNumber(),
+        maxCrypto: Number(account.formattedBalance),
+    };
+
+    const [state, setState] = useState<ReturnType<typeof useStakeEthState>>(undefined);
+
+    const { saveDraft, getDraft, removeDraft } = useFormDraft<StakeEthFormState>('stake-eth');
+    const draft = getDraft(account.key);
+    const isDraft = !!draft;
+
+    const { defaultValues } = useStakeEthFormDefaultValues(state?.formValues?.outputs[0].address);
+
+    // throttle initial state calculation
+    const initState = useStakeEthState(selectedAccount, fees, !!state, defaultValues);
+
+    useEffect(() => {
```

Comment:
This is probably not correct for staking

Reply:
Could you, please, elaborate what exactly is incorrect? Do you mean `useStakeEthState()` shouldn’t be used? Or do you mean the async setting of the state in the `useEffect()`? Or maybe both? :)

Removed both for now.

https://github.com/everstake/trezor-suite/pull/3/commits/b7a014ce3da889993ab618e180a812ae6fa172f1